src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -0,0 +1,247 @@
+                }
+                match &options.introspection_interval {
+                    Set(ii) => logging.interval = ii.0,
+                    Reset => logging.interval = Some(DEFAULT_REPLICA_LOGGING_INTERVAL),
+                    Unchanged => {}
+                }
+                match &options.replication_factor {
+                    Set(rf) => *replication_factor = *rf,
+                    Reset => *replication_factor = 1,
+                    Unchanged => {}
+                }
+                match &options.schedule {
+                    Set(new_schedule) => {
+                        *schedule = new_schedule.clone();
+                    }
+                    Reset => *schedule = Default::default(),
+                    Unchanged => {}
+                }
+                if !matches!(options.replicas, Unchanged) {
+                    coord_bail!("Cannot change REPLICAS of managed clusters");
+                }
+            }
+            Unmanaged => {
+                use AlterOptionParameter::*;
+                if !matches!(options.size, Unchanged) {
+                    coord_bail!("Cannot change SIZE of unmanaged clusters");
+                }
+                if !matches!(options.availability_zones, Unchanged) {
+                    coord_bail!("Cannot change AVAILABILITY ZONES of unmanaged clusters");
+                }
+                if !matches!(options.introspection_debugging, Unchanged) {
+                    coord_bail!("Cannot change INTROSPECTION DEGUBBING of unmanaged clusters");
+                }
+                if !matches!(options.introspection_interval, Unchanged) {
+                    coord_bail!("Cannot change INTROSPECTION INTERVAL of unmanaged clusters");
+                }
+                if !matches!(options.replication_factor, Unchanged) {
+                    coord_bail!("Cannot change REPLICATION FACTOR of unmanaged clusters");
+                }
+            }
+        }
+
+        if new_config == config {
+            return Ok(StageResult::Response(ExecuteResponse::AlteredObject(
+                ObjectType::Cluster,
+            )));
+        }
+
+        match (&config.variant, new_config.variant) {
+            (Managed(config), Managed(new_config)) => {
+                self.sequence_alter_cluster_managed_to_managed(
+                    Some(session),
+                    cluster_id,
+                    config,
+                    new_config,
+                    ReplicaCreateDropReason::Manual,
+                )
+                .await?;
+            }
+            (Unmanaged, Managed(new_config)) => {
+                self.sequence_alter_cluster_unmanaged_to_managed(
+                    session, cluster_id, new_config, options,
+                )
+                .await?;
+            }
+            (Managed(_), Unmanaged) => {
+                self.sequence_alter_cluster_managed_to_unmanaged(session, cluster_id)
+                    .await?;
+            }
```

Comment:
right now these all still happen on the main coordinator thread. We can easily move them off-thread now by spawning them in a tokio task and returning a handle to that task, is that desirable?

Reply:
right now I think no?

I believe we only want to do that for alter_managed_to_managed if performing a "graceful" reconfig, although we could potentially kick all of these off on a thread and return with StageResult::Handle or HandleRetire.

Reply:
If these do any long running async work like reach out to kubernetes and waiting for it to create resources then it might be nice to move them off the coord thread, but def not blocking this PR since we're maintaining the status quo here

Reply:
That requires refactoring the orchestrator/compute controller which few people own or understand. Definitely the right call to merge this, which puts us into position to incrementally improve this later on.